src/core/config-schema.helper.ts

```diff
@@ -0,0 +1,125 @@
+import * as Joi from '@hapi/joi';
+import chalk from 'chalk';
+
+const schema = Joi.object({
+  browserWidth: Joi.number()
+    .min(100)
+    .max(2000)
+    .required(),
+  browserHeight: Joi.number()
+    .min(100)
+    .max(2000)
+    .required(),
+  browserLanguage: Joi.string().required(),
+  timeout: Joi.number()
+    .min(1)
+    .required(),
+  elementsVisibilityTimeout: Joi.number()
+    .min(1)
+    .required(),
+  waitForPageTimeout: Joi.number()
+    .min(1)
+    .required(),
+  downloadTimeout: Joi.number()
+    .min(1)
+    .required(),
+  reports: Joi.string().required(),
+  downloads: Joi.string().required(),
+  data: Joi.string().required(),
+  features: Joi.array()
+    .items(Joi.string())
+    .required(),
+  pages: Joi.array()
+    .items(Joi.string())
+    .required(),
+  matchers: Joi.array()
+    .items(Joi.string())
+    .required(),
+  generators: Joi.array()
+    .items(Joi.string())
+    .required(),
+  form_handlers: Joi.array()
+    .items(Joi.string())
+    .required(),
+  step_definitions: Joi.array()
+    .items(Joi.string())
+    .required(),
+  comparators: Joi.array()
+    .items(Joi.string())
+    .required(),
+  dictionaries: Joi.array()
+    .items(Joi.string())
+    .required(),
+  transformers: Joi.array()
+    .items(Joi.string())
+    .required(),
+  regexes: Joi.array()
+    .items(Joi.string())
+    .required(),
+  hooks: Joi.array()
+    .items(Joi.string())
+    .required(),
+  imageComparator: {
+    baselineFolder: Joi.string().required(),
+    temporaryFolder: Joi.string().required(),
+    saveAboveTolerance: Joi.number()
+      .min(1)
```

Comment:
Why 0 is not included? It should be from 0 to 100

Reply:
0 will be OK but, I don't agree with 100 because it is ```saveAboveTolerance``` we cannot have deviation tolerance above 100%